### PR TITLE
Bodies, corpses, cadavers

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
```SUMMARY: Content "Added new 'corpses' map extra."```

#### Purpose of change
Add some more evidences that lots of people died prior to and during the Cataclysm.

#### Describe the solution
- Added "corpses" map extra that spawns up to 5 human bodies in random places with a puddles of blood around them. Additionally there is a 10% chance to spawn one fully torn human corpse with gibs around it, and a flock of stray dogs that actually did that to the corpse.
- Added `GROUP_STRAY_DOGS` monster groups with all dogs variations.
- Added `remains_human_generic` item group consisting of human flesh, bones and stomach.
- Added chance to spawn one human corpse in the fields, forests and on the road as part of `field`,  `forest`, and `road` item groups respectfully. The chance to spawn in all item groups is rather high, so corpses should spawn frequently. All corpses will be supplied with torn clothes and 30% of them will be supplied with everyday items.

#### Additional context
For some reason dogs shit all over the place immediately after spawning, I couldn't prevent this.
изображение

### CODE REVIEW

**Comment by FrozenSnowFox** [comment]
Would it be possible to jsonize this sort of map extra creation?  These map extras you've been adding are nice little additions that add a lot to the game and having a massive amount of variety would be great.  If it was possible to do in json it would be a nice simple entry for people interested in contributing.

File: src/map_extras.cpp [hunk 1]
```diff
@@ -2625,6 +2625,39 @@ static void mx_looters( map &m, const tripoint &abs_sub )
...
+    //Spawn up to 5 human corpses in random places
+    for( int i = 0; i < num_corpses; i++ ) {
+        const tripoint corpse_location = { rng( 1, SEEX * 2 - 1 ), rng( 1, SEEY * 2 - 1 ), abs_sub.z };
+        if( g->is_empty( corpse_location ) ) {
+            m.add_field( corpse_location, fd_blood, rng( 1, 3 ) );
+            m.add_corpse( corpse_location );
+            //50% chance to spawn blood in every tile around every corpse in 1-tile radius
+            for( const auto &loc : m.points_in_radius( corpse_location, 1 ) ) {
+                if( one_in( 2 ) ) {
+                    m.add_field( { loc.xy(), abs_sub.z }, fd_blood, rng( 1, 3 ) );
```
Comment (BevapDin) [inline src/map_extras.cpp]: Why can't you just use `loc` (which happens to be a `tripoint`) directly? It should already have the same z-value as the input point (`corpse_location`), which is the same value as the one used here to create a new tripoint.
```suggestion
                    m.add_field( loc, fd_blood, rng( 1, 3 ) );
```

**Comment by anothersimulacrum** [comment]
@FrozenSnowFox Pretty sure it already has been, see #31682

**Comment by LaVeyanFiend** [comment]
Oh boi, cannibal's getting a buff!

**Comment by SomeDeadGuy** [comment]
Maybe add new corpses with new ID's? Like additional variants of dead bodies?
Like headless corpse, corpse without the head, child corpse, adult corpse, pile of gore, skeleton covered in goo, burned up corpse. Can even make like untold stories based on the state of the corpse. 
I would like to present this in tileset too, but now all I have is like one corpse graphics for humans and it is really limiting. With different IDs I can add variety to the look of the game.

File: src/map_extras.cpp (hunk 1 above)
Comment (Night-Pryanik) [reply src/map_extras.cpp]: Will fix this shortly, thanks for the review.

**Comment by Night-Pryanik** [comment]
I guess this would be a great addition to the game, though it's out of scope of this PR.
I'll try to implement it in other PR.

**Comment by FrozenSnowFox** [comment]
> @FrozenSnowFox Pretty sure it already has been, see #31682

Ah my mistake, thanks for the link and sorry for derailing.

**Comment by Night-Pryanik** [comment]
@anothersimulacrum well, while now we can use various mapgen functions in json, the very mapgen extra mapgen is still hardcoded and have to be written in C++ beforehand.

**Comment by ZhilkinSerg** [comment]
> 
> 
> @anothersimulacrum well, while now we can use various mapgen functions in json, the very mapgen extra mapgen is still hardcoded and have to be written in C++ beforehand.

One can use non-builtin functions (`mapgen` and `mapgen_update`) which can be written in json, though there would be less possibilities compared to extra functions written in C++ and there is also some bug (#33640).

File: src/map_extras.cpp [hunk 2]
```diff
@@ -2625,6 +2625,39 @@ static void mx_looters( map &m, const tripoint &abs_sub )
...
+    }
+    //10% chance to spawn a flock of stray dogs feeding on human flesh
+    if( one_in( 10 ) && num_corpses <= 4 ) {
+        const tripoint corpse_location = { rng( 1, SEEX * 2 - 1 ), rng( 1, SEEY * 2 - 1 ), abs_sub.z };
+        m.spawn_items( corpse_location, gibs );
+        m.add_field( corpse_location, fd_gibs_flesh, rng( 1, 3 ) );
+        //50% chance to spawn gibs and dogs in every tile around what's left of human corpse in 1-tile radius
+        for( const auto &loc : m.points_in_radius( corpse_location, 1 ) ) {
+            if( one_in( 2 ) ) {
+                m.add_field( { loc.xy(), abs_sub.z }, fd_gibs_flesh, rng( 1, 3 ) );
```
Comment (BevapDin) [inline src/map_extras.cpp]: Same here.

**Comment by kilozombie** [comment]
Could I ask how this fits into the lore?

It's been a while, and I heard there's been a lore rewrite, but my knowledge of Cataclysm is that a vast majority of the human population became Z because the blob was in the water grid long before Hour One. The only reason that scientist, military, drug dealer corpse map extras existed was because they were carrying bottled water exempt from the blob that most other people ingested.

That's why there are hundreds of Zs in town-- it didn't matter how you died, you were already fated to become one.

**Comment by Night-Pryanik** [comment]
> The only reason that scientist, military, drug dealer corpse map extras existed was because they were carrying bottled water exempt from the blob that most other people ingested.

Let's presume these people too.

After we implement humans becoming zombies after death, that would be a non-issue.

**Pull request merged**